cmd/uzfs/uzfs.c

```diff
@@ -1837,6 +1841,142 @@ uzfs_object_test(int argc, char **argv)
 	return (0);
 }
 
+typedef struct bench_thread_ctx {
+	libuzfs_dataset_handle_t *dhp;
+	uint64_t obj;
+	uint64_t blksize;
+	uint64_t total_size;
+	boolean_t write;
+	pthread_t tid;
+} bench_thread_ctx_t;
+
+static void *worker_func(void *arg) {
+	bench_thread_ctx_t *ctx = arg;
+
+	boolean_t sync = B_FALSE;
+	char *buf = umem_alloc(ctx->blksize, UMEM_NOFAIL);
+	memset(buf, 0xf1, ctx->blksize);
+
+	uint64_t nwritten = 0;
+
+	while (nwritten < ctx->total_size) {
+		if (ctx->write) {
+			VERIFY0(libuzfs_object_write(ctx->dhp,
+			    ctx->obj, nwritten, ctx->blksize, buf, sync));
+		} else {
+			VERIFY3U(libuzfs_object_read(ctx->dhp, ctx->obj,
+			    nwritten, ctx->blksize, buf), ==, ctx->blksize);
+			VERIFY((uint8_t)buf[101] == 0xf1);
+		}
+		nwritten += ctx->blksize;
+	}
+
+	umem_free(buf, ctx->blksize);
+	return (NULL);
+}
+
+static uint64_t
+micros_elapsed(struct timeval *before, struct timeval *after)
+{
+	return (after->tv_sec - before->tv_sec) * 1000000 +
+	    after->tv_usec - before->tv_usec;
+}
+
+static int
+uzfs_io_bench(int argc, char **argv)
+{
+	int nthread = atoi(argv[1]);
+
+	argc -= 2;
+	argv += 2;
+
+	uint64_t total_megas = 4096;
```

Comment:
should we expose this as a parameter? so that the user can control if memory cache hit when test read